src/main/java/com/flightstats/hub/config/server/JettyServer.java

```diff
@@ -0,0 +1,145 @@
+        serverConnector.setIdleTimeout(systemProperties.getHttpIdleTimeInMillis());
+        server.setConnectors(new Connector[]{serverConnector});
+
+        // build Jersey HTTP context
+        ResourceConfig resourceConfig = buildResourceConfig(injector);
+        JettyHttpContainer httpContainer = ContainerFactory.createContainer(JettyHttpContainer.class, resourceConfig);
+
+        // build Jetty WebSocket context
+        ServletContextHandler wsContext = new ServletContextHandler(server, "/", ServletContextHandler.SESSIONS);
+        wsContext.setContextPath("/");
+        ServerContainer wsContainer = WebSocketServerContainerInitializer.configureContext(wsContext);
+        wsContainer.addEndpoint(WebSocketChannelEndpoint.class);
+        wsContainer.addEndpoint(WebSocketDayEndpoint.class);
+        wsContainer.addEndpoint(WebSocketHourEndpoint.class);
+        wsContainer.addEndpoint(WebSocketMinuteEndpoint.class);
+        wsContainer.addEndpoint(WebSocketSecondEndpoint.class);
+        wsContainer.addEndpoint(WebSocketHashEndpoint.class);
+
+        // use handler collection to choose the proper context
+        HttpAndWSHandler handler = new HttpAndWSHandler(metricsRequestFilter);
+        handler.addHttpHandler(httpContainer);
+        handler.addWSHandler(wsContext);
+        server.setHandler(handler);
+
+        // start everything up
+        server.start();
+
+        return server;
+    }
+
+    private ResourceConfig buildResourceConfig(Injector injector) {
+        ResourceConfig config = new ResourceConfig();
+        config.register(new GuiceToHK2Adapter(injector));
+        config.register(new ObjectMapperResolver(HubBindings.objectMapper()));
+        config.register(JacksonJsonProvider.class);
+        config.registerClasses(
+                CORSFilter.class,
+                EncodingFilter.class,
+                StreamEncodingFilter.class,
+                GZipEncoder.class,
+                DeflateEncoder.class
+        );
+        config.packages("com.flightstats.hub");
+        return config;
+    }
+
+    private SslContextFactory getSslContextFactory() throws IOException {
+
+        SslContextFactory sslContextFactory = null;
+        if (appProperties.isAppEncrypted()) {
+            log.info("starting hub with ssl!");
+
+            final String keyStorePath = appProperties.getKeyStorePath() + HubHost.getLocalName() + ".jks";
+            log.info("using key store path: {}", keyStorePath);
+
+            sslContextFactory = new SslContextFactory();
+            sslContextFactory.setKeyStorePath(keyStorePath);
+
+            String keyStorePasswordPath = appProperties.getKeyStorePasswordPath();
+            URL passwordUrl = new File(keyStorePasswordPath).toURI().toURL();
+            String password = Resources.readLines(passwordUrl, StandardCharsets.UTF_8).get(0);
+            sslContextFactory.setKeyStorePassword(password);
+        }
+        return sslContextFactory;
```

Comment:
I prefer to keep single point of exit in my methods.
I made this method more readable.

Reply:
This is a stylistic group choice, but FWIW, the single point of exit is an old guideline created back when methods tended to be much longer. These days, If a method is so long it's confusing as to where the points of exit are, then it's probably too long and needs some helper methods. For a method like this, it has two mutually exclusive code paths and I prefer to make it obvious with an else:
```
if (happy condition) {
    blah
    blah
    blah
    blah
    blah
    return happyThing; 
} else {
    // Initialization isn't relevant when server is in ZippyZoo configuration
    return null;
}
```

I save the if-without-else for cases where both the if and following clause both do useful work. Here, the scope of sslContextFactory makes it look like it has a use outside of the if statement, but it really doesn't since it's always null.